sdk/core/System.ClientModel/tests/gen.unit/ContextGeneratorTests.cs

```diff
@@ -1140,5 +1140,456 @@ public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+            var depResult = CompilationHelper.RunSourceGenerator(depCompilation, out var newDepCompilation);
+
+            Assert.IsNotNull(depResult.GenerationSpec);
+            Assert.AreEqual("LocalContext", depResult.GenerationSpec!.Type.Name);
+            Assert.AreEqual("TestDependency", depResult.GenerationSpec.Type.Namespace);
+            Assert.AreEqual(0, depResult.Diagnostics.Length);
+            Assert.AreEqual("public", depResult.GenerationSpec!.Modifier);
+            Assert.AreEqual(1, depResult.GenerationSpec.TypeBuilders.Count);
+
+            string source =
+$$"""
+  using System;
+  using System.ClientModel.Primitives;
+  using TestDependency;
+
+  namespace TestProject
+  {
+  #pragma warning disable TEST001 // Experimental type
+      [ModelReaderWriterBuildable(typeof(JsonModel))]
+  #pragma warning restore TEST001 // Experimental type
+      public partial class MyLocalContext : ModelReaderWriterContext { }
+
+      public class Caller
+      {
+          public void Call()
+          {
+  #pragma warning disable TEST001 // Experimental type
+              ModelReaderWriter.Read<JsonModel>(BinaryData.Empty, ModelReaderWriterOptions.Json, MyLocalContext.Default);
+  #pragma warning restore TEST001 // Experimental type
+          }
+      }
+  }
+  """;
+
+            Compilation compilation = CompilationHelper.CreateCompilation(
+                source,
+                additionalReferences: [newDepCompilation.ToMetadataReference()],
+                contextName: "MyLocalContext");
+
+            var result = CompilationHelper.RunSourceGenerator(compilation, out var newCompilation);
+
+            Assert.IsNotNull(result.GenerationSpec);
+            Assert.AreEqual("MyLocalContext", result.GenerationSpec!.Type.Name);
+            Assert.AreEqual("TestProject", result.GenerationSpec.Type.Namespace);
+            Assert.AreEqual(0, result.Diagnostics.Length);
+            Assert.AreEqual("public", result.GenerationSpec!.Modifier);
+            Assert.AreEqual(1, result.GenerationSpec.TypeBuilders.Count);
+            Assert.AreEqual(1, result.GenerationSpec.ReferencedContexts.Count);
+
+            var myLocalContext = newCompilation.GetTypeByMetadataName("TestProject.MyLocalContext");
+            Assert.IsNotNull(myLocalContext, "MyLocalContext should not be null.");
+
+            var referenceContextFiled = myLocalContext!.GetMembers("s_referenceContexts")
+                .OfType<IFieldSymbol>()
+                .FirstOrDefault(f => f.IsStatic);
+            Assert.IsNotNull(referenceContextFiled, "s_referenceContexts field should not be null.");
+        }
+
+        [TestCase("JsonModel[]")]
+        [TestCase("List<JsonModel>")]
+        [TestCase("Dictionary<string, JsonModel>")]
+        [TestCase("ReadOnlyMemory<JsonModel>")]
+        [TestCase("JsonModel[][]")]
+        [TestCase("JsonModel[,]")]
+        public void BuilderForExperimentalCollectionHasSuppression(string collectionType)
```

Comment:
Can you add verification that the source text for the builders have the suppress like we did in the other tests?  There are N builders for each of these tests and each of them should have the suppress since they would reference the experimental type.